### PR TITLE
feat: add zstd documentation

### PR DESCRIPTION
**Pull request by Michaelhobo**
Changelog: Add zstd documentation
Ticket: None

Signed-off-by: Michael Ho <callmemikeh@gmail.com>

### CODE REVIEW

File: .ispell_english [hunk 1]
```diff
@@ -2538,8 +2538,8 @@ yo
 yocto
 yoctoproject
 yongjhih
-You�
-Your�
+You�
+Your�
```
Comment (kacf) [inline .ispell_english:2542]: Looks like the editor messed up some unrelated words here. It has replaced some accented letters with the 0xFFFD "unknown" UTF code sequence. TBH I don't know what those accents are doing here in the first place, but 0xFFFD is definitely even more incorrect, so best to remove those.

File: 04.System-updates-Debian-family/99.Variables/docs.md [hunk 2]
```diff
@@ -64,11 +64,16 @@ it is missing a "MENDER_" prefix.
...
 
 The compression algorithm to use when generating the Artifact. In general LZMA
 will produce a smaller Mender Artifact (2-3x) but will significantly increase
-time spent generating the Mender Artifact (10x).
+time spent generating the Mender Artifact (10x), compared to gzip.
+
+Zstd provides different compression levels, and can achieve comparable
+compression ratios to both gzip and LZMA, at faster compression and
+decompression throughput.
+See https://engineering.fb.com/2018/12/19/core-data/zstandard for details.
```
Comment (kacf) [inline 04.System-updates-Debian-family/99.Variables/docs.md:76]: Can you perhaps mention which of the three variants above is the default if you just pick the short one?

**Review by kacf — CHANGES_REQUESTED** [review]
Thanks @Michaelhobo

File: .ispell_english (hunk 1 above)
Comment (Michaelhobo) [reply .ispell_english]: Fixed. That was weird, my editor didn't even show these changes

File: 04.System-updates-Debian-family/99.Variables/docs.md (hunk 2 above)
Comment (Michaelhobo) [reply 04.System-updates-Debian-family/99.Variables/docs.md:76]: There's no longer a short one, it's been renamed to `zstd_fast`

**Review by kacf — APPROVED** [review]
Thanks! I will merge this once the code PRs are merged.

**Pull request merged**